src/stack/marshalling-yard.c

```diff
@@ -0,0 +1,113 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "stack.h"
+
+void append(char* ch, int* capasity, int* lenght, char** str)
+{
+    if (*capasity - 1 <= *lenght)
+    {
+        char* ptr = malloc(sizeof(char) * *capasity * 2);
+        if (ptr == NULL)
+            return;
+
+        *capasity *= 2;
+        strcpy(ptr, *str);
+        free(*str);
+        *str = ptr;
+    }
+    (*str)[*lenght] = *ch;
+    (*lenght)++;
+    (*str)[*lenght] = '\0';
+}
+
+char* reading(int* capasity, int* lenght)
+{
+    char* str = malloc(sizeof(char) * *capasity);
+    if (str == NULL)
+        return NULL;
+    char ch;
+    while ((ch = getchar()) != EOF)
+        append(&ch, capasity, lenght, &str);
+    return str;
+}
+
+int operand(char symbol)
+{
+    if (symbol == '+' || symbol == '*' || symbol == '-' || symbol == '/')
+        return 1;
+    return 0;
+}
+
+int digit(char symbol)
+{
+    if (operand(symbol) == 0 && symbol != ')' && symbol != '(')
+        return 1;
+    return 0;
+}
+
+int operand_comparison(char symbol)
```

Comment:
Это не операнд, а оператор. И функция не сравнивает какие-то операторы, а вычисляет приоритет одного оператора, так что
```suggestion
int operator_priority(char symbol)
```